File: server/src/main/java/au/com/codeka/warworlds/server/util/CookieHelper.java
package au.com.codeka.warworlds.server.util;

import java.security.SecureRandom;

/**
 * Helper for working with cookies.
 */
public class CookieHelper {

  private static final char[] COOKIE_CHARS =
      "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789".toCharArray();
  private static final int COOKIE_LENGTH = 40;

  public static String generateCookie() {
    return generateCode(COOKIE_CHARS, COOKIE_LENGTH);
  }

  /** Generates a cookie, which is basically just a long-ish string of random bytes. */
  private static String generateCode(char[] alphabet, int length) {
    // generate a random string for the session cookie
    SecureRandom rand = new SecureRandom();
    StringBuilder cookie = new StringBuilder();
    for (int i = 0; i < length; i++) {
      cookie.append(alphabet[rand.nextInt(alphabet.length)]);
    }

    return cookie.toString();
  }

}

File: server/src/main/kotlin/au/com/codeka/warworlds/server/util/CookieHelper.kt
package au.com.codeka.warworlds.server.util

import java.security.SecureRandom

/**
 * Helper for working with cookies.
 */
object CookieHelper {
  private val COOKIE_CHARS = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789".toCharArray()
  private const val COOKIE_LENGTH = 40
  fun generateCookie(): String {
    return generateCode(COOKIE_CHARS, COOKIE_LENGTH)
  }

  /** Generates a cookie, which is basically just a long-ish string of random bytes.  */
  private fun generateCode(alphabet: CharArray, length: Int): String {
    // generate a random string for the session cookie
    val rand = SecureRandom()
    val cookie = StringBuilder()
    for (i in 0 until length) {
      cookie.append(alphabet[rand.nextInt(alphabet.size)])
    }
    return cookie.toString()
  }
}